src/game.zig

```diff
@@ -765,6 +768,117 @@ pub fn hyperSpeedToggle() void {
+	const maxY: i32 = @intFromFloat(@floor(skewedBox.max[swizzle[1]]));
+	const uniformZ: i32 =
+		if (uniformMin) @intFromFloat(@floor(skewedBox.min[swizzle[2]]))
+		else @intFromFloat(@floor(skewedBox.max[swizzle[2]]));
+
+	var posX: i32 = minX;
+	while (posX <= maxX) : (posX += 1) {
+		var posY: i32 = minY;
+		while (posY <= maxY) : (posY += 1) {
+			var blockPos: Vec3i = undefined;
+			blockPos[swizzle[0]] = posX;
+			blockPos[swizzle[1]] = posY;
+			blockPos[swizzle[2]] = uniformZ;
+			const block =
+				if (side == .client) main.renderer.mesh_storage.getBlock(blockPos[0], blockPos[1], blockPos[2])
+				else main.server.world.?.getBlock(blockPos[0], blockPos[1], blockPos[2]);
+			if (block != null and isBlockIntersecting(block, blockPos[0], blockPos[1], blockPos[2], center, extent)) {
+				block.?.onEntityTouching(entity, blockPos[0], blockPos[1], blockPos[2], neighbor);
+			}
+		}
+	}
+}
+
+fn touchBlocks(entity: main.server.Entity, hitBox: collision.Box, comptime side: main.utils.Side) void {
+	const boundingBox: collision.Box = .{
+		.min = entity.pos + hitBox.min + touchRetraction,
+		.max = entity.pos + hitBox.max - touchRetraction
+	};
+	touchNeighborBlocks(entity, boundingBox, side, Vec3i {0, 1, 2}, false, chunk.Neighbor.dirUp);
+	touchNeighborBlocks(entity, boundingBox, side, Vec3i {0, 1, 2}, true, chunk.Neighbor.dirDown);
+	touchNeighborBlocks(entity, boundingBox, side, Vec3i {1, 2, 0}, false, chunk.Neighbor.dirPosX);
+	touchNeighborBlocks(entity, boundingBox, side, Vec3i {1, 2, 0}, true, chunk.Neighbor.dirNegX);
+	touchNeighborBlocks(entity, boundingBox, side, Vec3i {0, 2, 1}, false, chunk.Neighbor.dirPosY);
+	touchNeighborBlocks(entity, boundingBox, side, Vec3i {0, 2, 1}, true, chunk.Neighbor.dirNegY);
+}
+
+fn insideBlocks(entity: main.server.Entity, hitBox: collision.Box, comptime side: main.utils.Side) void {
+	const boundingBox: collision.Box = .{
+		.min = entity.pos + hitBox.min + touchRetraction,
+		.max = entity.pos + hitBox.max - touchRetraction
+	};
+
+	const center = boundingBox.center();
+	const extent = boundingBox.extent();
+
+	const minX: i32 = @intFromFloat(@floor(boundingBox.min[0]));
+	const maxX: i32 = @intFromFloat(@floor(boundingBox.max[0]));
+	const minY: i32 = @intFromFloat(@floor(boundingBox.min[1]));
+	const maxY: i32 = @intFromFloat(@floor(boundingBox.max[1]));
+	const minZ: i32 = @intFromFloat(@floor(boundingBox.min[2]));
+	const maxZ: i32 = @intFromFloat(@floor(boundingBox.max[2]));
+
+	var posX: i32 = minX;
+	while (posX <= maxX) : (posX += 1) {
+		var posY: i32 = minY;
+		while (posY <= maxY) : (posY += 1) {
+			var posZ: i32 = minZ;
+			while (posZ <= maxZ) : (posZ += 1) {
+				const block =
+					if (side == .client) main.renderer.mesh_storage.getBlock(posX, posY, posZ)
+					else main.server.world.?.getBlock(posX, posY, posZ);
+				if (block != null and isBlockIntersecting(block, posX, posY, posZ, center, extent)) {
+					block.?.onEntityInside(entity, posX, posY, posZ);
+				}
+			}
+		}
+	}
+}
+
```

Comment:
Could you please move all the collision checking code into the collision namespace, next to other collision-related functions?